sqlglot/dialects/snowflake.py

```diff
@@ -601,8 +601,8 @@ def _parse_table_parts(
                 file_format = None
                 pattern = None
 
-                self._match(TokenType.L_PAREN)
-                while self._curr and not self._match(TokenType.R_PAREN):
+                wrapped = self._match(TokenType.L_PAREN)
+                while self._curr and (wrapped and not self._match(TokenType.R_PAREN)):
```

Comment:
i don't think you need the parenthesis

self._curr and wrappend and not self._match right?

Reply:
Right, shouldn't make a difference. I usually wrap such paired bools to put a mental (and future) note when re-reading these chained conditions. Do we want to preserve the most simplified version of branches in general?

Reply:
yea let's simplify